module_utils/ansible_nailgun_cement.py

```diff
@@ -523,6 +529,15 @@ def find_product(module, name, organization, failsafe=False):
     return handle_find_response(module, product.search(), message="No product found for %s" % name, failsafe=failsafe)
 
 
+def find_products(module, names, organization, failsafe=False):
+    return list(map(lambda name: find_product(module, name, organization, failsafe=failsafe), names))
```

Comment:
I don't think this works as expected. It generates a bogus request which causes a 500 in Katello: https://paste.fedoraproject.org/paste/B5mlDe8K007CqoD~~Lovfg

Are you seeing this, too?

Reply:
I thought python lambdas bind to a local scope. I didn't think you could use local variables without explicitly binding them, like this:

```py
return list(map(lambda name, module=module, organization=organization: find_product(module, name, organization, failsafe=failsafe), names))
```

Reply:
Can you try with
```python
return [find_product(module, name, organization, failsafe=failsafe) for name in names]
```

Reply:
Seems like maybe an issue with `find_product`. Here's what I get with that change: https://paste.fedoraproject.org/paste/6yxWFOLMgkPPyG6w9pz6Sg

I think it's the nested `name` attribute: `Parameters: {"organization_id"=>2, "name"=>{"name"=>"custom"}, "api_version"=>"v2", "product"=>{"name"=>{"name"=>"custom"}, "organization_id"=>2}}`

Reply:
I have no idea, where the `"name"=>{"name"=>"custom"}` could come from, but yes, it looks weird.